### PR TITLE
QtWebEngine

### PR DESCRIPTION
**Pull request by romibi**
If QtWebEngine is available use that instead of QtWebKit.
Building with QtWebkit is still possible.

Introduces the Build Option "WITH_WEBENGINE"

Appveyor Builds:
* MSVC Builds with Qt5.6 with QtWebEngine (x64/x86)
* mingw Build with Qt5.6 but no WebPreview
* mingw Build with Qt5.5 with QtWebKit

(To build with QtWebEngine on Windows Visual Studio is needed)

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -94,7 +110,9 @@ build_script:
                             "bin\\libphononexperimental.*\.dll",
                             "bin\\phonon_backend\\.*",
                             "plugins\\phonon_backend\\.*",
-                            "plugins\\phonon4qt5_backend\\.*") @(
+                            "plugins\\phonon4qt5_backend\\.*")+$webPrevFiles
```
Comment (TheOneRing) [inline appveyor.yml]: I would move the append of webPrevFiles to an extra line, for better readability.

`$whitelist += $webPrevFiles `

**Comment by digitalcircuit** [comment]
Asides from the few minor tweaks remarked above, looks good.  Thank you for your effort to make web previews work again!

As to Visual Studio being required for building (an unfortunate limitation of Qt / Chromium), maybe a future pull request can drop any unneeded MingW builds after this has some time in master.

File: CMakeLists.txt [hunk 2]
```diff
@@ -76,7 +76,10 @@ if (NOT WITH_KDE)
 endif()
 
 # For this, the feature info is added after we know if QtWebkit is installed
-option(WITH_WEBKIT "WebKit support (for link previews)" ON)
+option(WITH_WEBKIT "WebKit support (for link previews) (legacy)" ON)
+
+# For this, the feature info is added after we know if QtWebkit is installed
+option(WITH_WEBENGINE "WebKit support (for link previews)" ON)
```
Comment (digitalcircuit) [inline CMakeLists.txt]: Minor copy-paste matter: "WebKit" should be corrected to "WebEngine" in the comment (line 81) and the description.

*Pardon the double-comment; I didn't realize Github treated comments differently depending on if you're in the ```Files changed ``` tab versus commenting directly on commits.*

**Comment by romibi** [comment]
Added QtWebEngine logic to the Mac-Deploy-Script ...
That was more difficult than expected ... thou the solution looks simple ...

(wow QtWebEngine on OSX is quite big)

**Comment by romibi** [comment]
Appveyor failed only because of failed NSIS download: [see Build Log Line 200](https://ci.appveyor.com/project/Quassel/quassel/build/231/job/9no9rde7hev17aba#L200)

**Pull request closed without merging**